db/migrate/20150925165323_devise_create_users.rb

```diff
@@ -0,0 +1,42 @@
+class DeviseCreateUsers < ActiveRecord::Migration
+  def change
+    create_table(:users) do |t|
+      t.timestamps null: false
+      ## Database authenticatable
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.inet     :current_sign_in_ip
+      t.inet     :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+      t.boolean :verified, null: false, default: false
+      t.string :type
```

Comment:
I'm personally not a huge fan of STI anymore for logging in — I've found that it hurts more often than helps, especially if you think that either the `Student` or `Admin` model will need more columns as time goes on. In fact, I'd even make the stronger statement of using STI if and only if you don't think `Student` and `Admin` won't need additional columns in the future. Since this also includes foreign keys (eg. `belongs_to` relationships) I don't think you'll want to use STI in this case since you'll end up with a bunch of columns on `Admin` that don't do anything as well.